### PR TITLE
Add GPQA Diamond and fix evaluation deps

### PR DESCRIPTION
**Pull request by lewtun**
Adds GPQA diamond and various important fixes for evaluation (parsing & incompat between latest `vllm` and `lighteval`). I've also unified the Slurm scripts for evaluation so we don't have multiple ways to eval models.

# TODO

- [x] Update Slurm script to disable system prompt for GPQA
- [x] Rerun evals once #188 and #193 are merged and env is updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,16 +134,33 @@ We use `lighteval` to evaluate models, with custom tasks defined in `src/open_r1
...
+    --custom-tasks src/open_r1/evaluate.py \
+    --use-chat-template \
+    --output-dir $OUTPUT_DIR
+
+# GPQA Diamond
+TASK=gpqa:diamond
 lighteval vllm $MODEL_ARGS "custom|$TASK|0|0" \
     --custom-tasks src/open_r1/evaluate.py \
     --use-chat-template \
-    --system-prompt="Please reason step by step, and put your final answer within \boxed{}." \
```
Comment (lewtun) [inline README.md:143]: Not needed for the DeepSeek models (gives ~1 point gain if included)

File: README.md [hunk 2]
```diff
@@ -56,17 +56,17 @@ uv venv openr1 --python 3.11 && source openr1/bin/activate && uv pip install --u
...
-pip install vllm>=0.7.0 --extra-index-url https://download.pytorch.org/whl/cu121
+uv pip install vllm==0.7.1 --extra-index-url https://download.pytorch.org/whl/cu121 --index-strategy unsafe-best-match
 export LD_LIBRARY_PATH=$(python -c "import site; print(site.getsitepackages()[0] + '/nvidia/nvjitlink/lib')"):$LD_LIBRARY_PATH
 ```
 
 This will also install PyTorch `v2.5.1` and it is **very important** to use this version since the vLLM binaries are compiled for it. You can then install the remaining dependencies for your specific use case via `pip install -e .[LIST OF MODES]`. For most contributors, we recommend:
 
 ```shell
-pip install -e ".[dev]"
+GIT_LFS_SKIP_SMUDGE=1 uv pip install -e ".[dev]" --link-mode=copy
```
[the remark this answers is not in the document]
Comment (edbeeching) [reply README.md:69]: Ah, I had this issue, I had reverted back to pip, glad you fixed it.

File: slurm/evaluate.slurm [hunk 3]
```diff
@@ -1,55 +1,75 @@
...
-MODEL=$1
-TASK=$2
-# Check if a third argument is passed, if it is tp then eval with tensor parallelism. Required for larger models
-if [ -n "$3" ] && [ "$3" == "tp" ]; then
-  MODEL_ARGS="pretrained=$MODEL,dtype=float16,tensor_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
+# Set Whether to use tensor parallelism or data parallelism
+if [ "$TENSOR_PARALLEL" = "True" ]; then
+    # use TP to shard model across NUM_GPUS
+    export VLLM_WORKER_MULTIPROC_METHOD=spawn
+    MODEL_ARGS="pretrained=$MODEL_ID,revision=$MODEL_REVISION,trust_remote_code=$TRUST_REMOTE_CODE,dtype=float16,tensor_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
```
Comment (edbeeching) [inline slurm/evaluate.slurm:26]:
```suggestion
    MODEL_ARGS="pretrained=$MODEL_ID,revision=$MODEL_REVISION,trust_remote_code=$TRUST_REMOTE_CODE,dtype=bfloat16,tensor_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
```

File: slurm/evaluate.slurm [hunk 4]
```diff
@@ -1,55 +1,75 @@
...
 else
-  MODEL_ARGS="pretrained=$MODEL,dtype=float16,data_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
+    MODEL_ARGS="pretrained=$MODEL_ID,revision=$MODEL_REVISION,trust_remote_code=$TRUST_REMOTE_CODE,dtype=float16,data_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
```
Comment (edbeeching) [inline slurm/evaluate.slurm:28]:
```suggestion
    MODEL_ARGS="pretrained=$MODEL_ID,revision=$MODEL_REVISION,trust_remote_code=$TRUST_REMOTE_CODE,dtype=bfloat16,data_parallel_size=$NUM_GPUS,max_model_length=32768,gpu_memory_utilisation=0.8"
```

File: setup.py [hunk 5]
```diff
@@ -53,17 +53,17 @@
     "huggingface-hub[cli]>=0.19.2,<1.0",
     "isort>=5.12.0",
     "liger_kernel==0.5.2",
-    "lighteval @ git+https://github.com/huggingface/lighteval.git@0e462692436e1f0575bdb4c6ef63453ad9bde7d4#egg=lighteval[math]",
-    "math-verify>=0.3.3",  # Used for math verification in grpo
+    "lighteval @ git+https://github.com/huggingface/lighteval.git@3c9b0c9dde6718b23ef5b0f4960355f0d494bdfc#egg=lighteval[math]",
```
Comment (lewtun) [inline setup.py:56]: Bump to latest commit once vllm fix for DDP is merged: https://github.com/huggingface/lighteval/pull/541

File: slurm/evaluate.slurm (hunk 4 above)
Comment (lewtun) [reply slurm/evaluate.slurm:28]: AFAIK, for inference it is better to use `float16` since you have a smaller range with high precision (i.e. better for the typical small numbers one has in logits)

**Review by edbeeching — APPROVED** [review]
